tests/repository/test_search_content.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from pubtools.pulplib import Repository, DetachedException
+
+
+def test_detached():
+    """search_content raises if called on a detached repo"""
+    with pytest.raises(DetachedException):
+        Repository(id="some-repo").search_content()
+
+
+class TestSearchContent(object):
+    def __init__(self, requests_mocker, client):
+        self.repo = Repository(id="some-repo")
+        self.repo.__dict__["_client"] = client
+
+        requests_mocker.post(
+            "https://pulp.example.com/pulp/api/v2/repositories/some-repo/search/units/",
+            json=[
+                {
+                    "_content_type_id": "iso",
+                    "name": "hello.txt",
+                    "size": 23,
+                    "checksum": "a" * 64,
+                },
+                {
+                    "_content_type_id": "srpm",
+                    "name": "bash",
+                    "epoch": "0",
+                    "filename": "bash-x86_64.srpm",
+                    "version": "4.0",
+                    "release": "1",
+                    "arch": "x86_64",
+                },
+                {
+                    "_content_type_id": "rpm",
+                    "name": "bash",
+                    "epoch": "0",
+                    "filename": "bash-x86_64.rpm",
+                    "version": "4.0",
+                    "release": "1",
+                    "arch": "x86_64",
+                },
+                {
+                    "_content_type_id": "modulemd",
+                    "name": "m1",
+                    "stream": "s1",
+                    "version": 1234,
+                    "context": "a1b2c3",
+                    "arch": "s390x",
+                },
+                {
+                    "_content_type_id": "modulemd_defaults",
```

Comment:
This isn't what a real modulemd_defaults unit looks like.  If you put in proper test data here, you should see a crash in schema checking (see earlier comment about wrong schema).